tests/org.eclipse.n4js.hlc.tests/src/org/eclipse/n4js/hlc/tests/InstallCompileRunN4jscExternalShadowNestedNodeModules.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2019 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.hlc.tests;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.eclipse.n4js.hlc.base.BuildType;
+import org.eclipse.n4js.hlc.base.ExitCodeException;
+import org.eclipse.n4js.hlc.base.N4jscBase;
+import org.eclipse.n4js.test.helper.hlc.N4CliHelper;
+import org.eclipse.n4js.utils.io.FileDeleter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Predicates;
+
+/**
+ * Testing the shadowing of npm packages.
+ * <p>
+ * Situation:
+ * <ul>
+ * <li>node_modules folder in yarn workspace root contains an npm package "dep".
+ * <li>nested node_modules folder in project "P" contains an npm package called "dep".
+ * </ul>
+ * Expected behavior:
+ * <ol>
+ * <li>the npm package in the yarn workspace root should be compiled and linked to all projects in the workspace.
+ * <li>the npm package in the nested node_modules folder should *NOT* be compiled.
+ * </ol>
+ * <p>
+ * The above situation will occur if there are version conflicts between the package.json files of the member projects
+ * in a yarn workspace. The expected behavior is in line with how the IDE behaves in the UI case.
+ */
+public class InstallCompileRunN4jscExternalShadowNestedNodeModules extends AbstractN4jscTest {
```

Comment:
Does the class name need to end with "Test"?

Reply:
oh yeah, good find! Will fix ...